api/filters/selection_criteria.go

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright 2018 The Service Manager Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package filters
+
+import (
+	"net/http"
+
+	"github.com/Peripli/service-manager/pkg/query"
+	"github.com/Peripli/service-manager/pkg/util"
+	"github.com/Peripli/service-manager/pkg/web"
+)
+
+const (
+	// CriteriaFilterName is the name of the criteria filter
+	CriteriaFilterName = "CriteriaFilter"
+)
+
+// SelectionCriteria is filter that configures selection criteria per request.
+type SelectionCriteria struct {
+}
+
+// Name implements the web.Filter interface and returns the identifier of the filter.
+func (*SelectionCriteria) Name() string {
+	return CriteriaFilterName
+}
+
+// Run represents the selection criteria middleware function that processes the request and configures the request-scoped selection criteria.
+func (l *SelectionCriteria) Run(req *web.Request, next web.Handler) (*web.Response, error) {
```

Comment:
Are we going to support field querying on all entity fields because right now this filter will try to build criterions for everything that has been provided in the query. We might or might not want to introduce a "per entity" field query whitelist that specifies on what can an entity/resource be field-queried. (that is unless we can come up with parsing logic that works for all cases

Reply:
Could be a part of a new PR. 
```golang
type Person struct{
    Name QueryableString `json:"name"`
    Age QueryableInt `son:"
}
```